scripts/bin/auto-mount

```diff
@@ -1,17 +1,52 @@
-#! /bin/sh
-
-sleep 5
-
-$SNAP/usr/local/bin/udisksctl monitor | grep --line-buffered " Device:" | while read -r -- device; do
-
-  echo "$SNAP_NAME.auto-mount found ${device}"
-  if echo $device| grep -q "[0-9]$"; then
-    DEV="$(echo $device | sed 's/^Device: //')"
-  fi
-  if [ -n "$DEV" ]; then
-    RES=`$SNAP/usr/local/bin/udisksctl mount --no-user-interaction --block-device "$DEV"`
-    # process returned value into the mount point string
-    RES2=`echo $RES | cut -f4 -d\  | cut -f1 -d.`
-    echo $RES2 > $WATCHED
-  fi
-done
+#!/bin/sh
+
+. "$SNAP/bin/common"
+
+init_mount_dir() {
+    _arg_dir="$1"
+
+}
+
+watch_devices() {
+    inotifywait --quiet --monitor --event "moved_to" --event "delete" /dev/disk/by-uuid | \
+        while read -r _parent _action _uuid; do
+            case "$_action" in
+                MOVED_TO) try_mount "$_parent" "$_uuid" ;;
+                DELETE)   rm_mount  "$_uuid" ;;
+            esac
+        done
+}
+
+try_mount() {
+    _arg_parent="${1%/}"
+    _arg_uuid="$2"
+
+    _source="${_arg_parent}/${_arg_uuid}"
+    _target="${_mount_dir}/${_arg_uuid}"
+
+    log "$SNAP_NAME.auto-mount found: ${_source}"
+
+    if ! mount "${_source}" "${_target}" >/dev/null 2>&1; then
+        log "WARN: Could not mount ${_source} to ${_target}"
+        return
+    fi
+
+    echo "${_target}" >> "${_watch_file}"
```

Comment:
So `>>` instead of `>` would mean that we support installing snaps from more than just one partition. It's not wrong, but I'm not certain that's what we want?

Reply:
I'm actually not sure there's a difference with a FIFO. When I originally wrote this, I wasn't thinking about the fact that writing to a FIFO blocks until read from, so I wanted to make sure that each line was appended. I think it's completely reasonable to switch this over to `>`.

Anyhow, try_mount() is always called with a single device as an argument, and every device that's newly populated in `/dev/disk/by-uuid` will be processed, so by default we are actually checking every partition.

Reply:
_it's a fifo_ 
Then this is fine and it's overwhelming pedantry.